hamilton/plugins/jupyter_magic.py

```diff
@@ -163,95 +194,160 @@ def determine_notebook_type() -> str:
 
 @magics_class
 class HamiltonMagics(Magics):
-    """Magics to facilitate Hamilton development in Jupyter notebooks"""
+    """Magics to facilitate interactive Hamilton development in notebooks."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.driver = None
+        self.driver_name = None
+        self.builder = None
+        self.external_modules = list()
+        self.final_vars = list()  # empty list will be converted to "all variables"
+
+        if not hasattr(self, "notebook_env"):
+            self.notebook_env = determine_notebook_type()
 
     @magic_arguments()  # needed on top to enable parsing
+    @argument("module_name", help="Name for the module defined in this cell.")
     @argument(
-        "-m", "--module_name", help="Module name to provide. Default is jupyter_module."
```

Comment:
we need `-m` support for backwards compatibility, right?

Reply:
There's a bit of a conflict here. I believe `module_name` should be a required argument, e.g. `%%cell_to_module joke`. Setting a name requires close to 0 efforts and improves readability.

Then, keeping `--module_name / -m` raises issues:
- your cell will fail anyway because it's missing the required arg. `%%cell_to_module -m joke` won't work.
- having both the required arg and `--module_name` is confusing. `%%cell_to_module something -m joke` is ambiguous

Reply:
we can't push a backwards incompatible API change.